ScreenManager.c

```diff
@@ -166,11 +166,10 @@ static inline bool drawTab(const int* y, int* x, int l, const char* name, bool c
    (*x)++;
    if (*x >= l)
       return false;
-   int nameLen = strlen(name);
-   int n = MINIMUM(l - *x, nameLen);
+   int nameWidth = (int)strnlen(name, (unsigned int)(l - *x));
```

Comment:
Probably should `assert(l > 0);`. Also the parameter for `l` does not include the space used for the `[` and `]` from the tab drawing.

```suggestion
   int nameWidth = (int)strnlen(name, (size_t)(l - *x));
```

Reply:
This assertion is probably not needed because of the `(*x >= l)` right above. It might not be obvious but `(*x < l)` → `(0 < l - *x)`.

Reply:
Well, `0 < l - *x` does not imply `l > 0`, as `*x >= 0` isn't checked either …

Reply:
Hmmm. Good catch. Although I believe the assertion should be a separate commit.

One more thing is I'm not sure if the compiler can catch that `l - *x > 0` and can optimise accordingly (on the bit extension). I am about to test that out and see whether GCC and Clang can optimise as expected or any of them has a bug.

Reply:
I think, that if compilers don't catch this micro-optimization, this doesn't hurt for this case much, as this routine isn't gonna be called thousands of times per second. Doesn't hurt to report those missed optimizations to compiler vendors though, but there's no need to explicitly make the code more complicated if they are missed.

Reply:
[Clang did miss this optimization](https://github.com/llvm/llvm-project/issues/142283).